tests/integration.rs

```diff
@@ -7719,6 +7719,173 @@ mod common_sequential {
 
         handle_child_output(r, &output);
     }
+
+    #[test]
+    fn test_snapshot_restore_pvpanic() {
+        _test_snapshot_restore_devices(true);
+    }
+    fn _test_snapshot_restore_devices(pvpanic: bool) {
```

Comment:
Please leave a blank line before this function.

It feels a bit weird that you have a dedicated function but only use it once. Given the name of this function, do you plan to add more devices in the future?

Reply:
Yes, I plan to add more devices in the future.

Reply:
A new question: what's the rationale behind adding a new function than to extend the existing test helper? There is already one in the test suite for snapshotting.

Reply:
I think they are in different scenes. `_test_snapshot_restore` ensures the **guest** can work normally after restoring. `_test_snapshot_restore_devices` ensures the **device** can work normally. And adding these codes will make the former bloated.

Reply:
Okay. That's fair.